internal/storage-service/service.go

```diff
@@ -206,6 +206,77 @@ func (s *Service) Update(ctx context.Context, req *pb.StorageUpdateRequest) (*pb
 	return &pb.StorageUpdateResponse{}, nil
 }
 
+// Delete deletes a storage
+func (s *Service) Delete(ctx context.Context, req *pb.StorageDeleteRequest) (*pb.StorageDeleteResponse, error) {
+	s.log.WithFields(logrus.Fields{
+		"StorageType": req.StorageType,
+		"SystemId":    req.SystemId,
+	}).Info("Serving delete storage request")
+
+	// Get the current list of registered storage systems
+	s.log.Debug("Getting existing storages")
+	existingStorages, err := s.kube.GetConfiguredStorage(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	s.log.Debug("Getting system type")
+	systemType, ok := existingStorages[req.StorageType]
+	if !ok {
+		return nil, fmt.Errorf("error: storage of type %s is missing", req.StorageType)
+	}
+
+	s.log.Debug("Check the requested system ID exists")
+	if _, systemIDExists := systemType[req.SystemId]; !systemIDExists {
+		return nil, fmt.Errorf("error: system with ID %s does not exist", req.SystemId)
+	}
+
+	delete(systemType, req.SystemId)
```

Comment:
Could you add a log statement above this line regarding updating the storage data? Something like this:

https://github.com/dell/karavi-authorization/blob/1c6ec905cfcbc5c035df599d91716fecf24c8ee8/internal/storage-service/service.go#L199